Claude - Saving new OAuth credentials on refresh fails

ClaudeBar version: 0.4.58
macOS version: 15.7.5

When using the Claude provider in API/OAuth mode I've found I seem to get logged out of both ClaudeBar and claude code after a few hours. Checking the logs this seems to be due to an error with the process to refresh the OAuth credentials
```
[2026-04-09T21:37:45.226Z] [INFO] [probes] Claude API: Token expired or expiring soon, refreshing...
[2026-04-09T21:37:45.795Z] [ERROR] [credentials] Failed to save Claude credentials to Keychain (exit code: 2)
[2026-04-09T21:37:45.796Z] [INFO] [probes] Claude API: Token refreshed successfully
```

Checking [the code](https://github.com/tddworks/ClaudeBar/blob/8b1728373ac343bd25978ef630bde9abafd2044b/Sources/Infrastructure/Claude/ClaudeCredentialLoader.swift#L257) for saving to the keychain, I think I can see the issue. If I run these `security` commands manually in a terminal, the delete works fine but the add fails with a usage message and exit code 2 (as seen in the log above).
```
Usage: add-generic-password [-h] [-a account] [-s service] [-w password] [options...] [-A|-T appPath] [keychain]
    -a  Specify account name (required)
...
```

I think it is clear that the `-a` argument is missing from the `add-generic-password` command. If I add this when running the commands manually it then succeeds. Checking how `claude` creates the keychain entry, the value of `-a` should be the local users account name (e.g. what `whoami` shows).

I think if this is updated in the code the refresh process will save to the keychain correctly and prevent these log outs I'm experiencing.

**Improvement**
Could potentially also add `-U` to `add-generic-password` which will update the existing entry and mean the `delete-generic-password` is no longer needed. 
```
    -U  Update item if it already exists (if omitted, the item cannot already exist)
```

So the final command would look like `/usr/bin/security add-generic-password -U -s 'Claude Code-credentials' -a <username> -w <JSON>`

--- a/Sources/Infrastructure/Claude/ClaudeCredentialLoader.swift
+++ b/Sources/Infrastructure/Claude/ClaudeCredentialLoader.swift
@@ -58,6 +58,20 @@ public struct ClaudeCredentialLoader: Sendable {
     /// Refresh buffer: 5 minutes before expiration
     private static let refreshBufferMs: Double = 5 * 60 * 1000
 
+    static func keychainSaveArguments(
+        service: String,
+        account: String,
+        password: String
+    ) -> [String] {
+        [
+            "add-generic-password",
+            "-U",
+            "-s", service,
+            "-a", account,
+            "-w", password
+        ]
+    }
+
     public init(
         homeDirectory: String = NSHomeDirectory(),
         keychainService: String = "Claude Code-credentials",
@@ -261,19 +275,13 @@ public struct ClaudeCredentialLoader: Sendable {
             return
         }
 
-        // Delete existing item first (ignore errors if not found)
-        let deleteProcess = Process()
-        deleteProcess.executableURL = URL(fileURLWithPath: "/usr/bin/security")
-        deleteProcess.arguments = ["delete-generic-password", "-s", keychainService]
-        deleteProcess.standardOutput = Pipe()
-        deleteProcess.standardError = Pipe()
-        try? deleteProcess.run()
-        deleteProcess.waitUntilExit()
-
-        // Add new item
         let addProcess = Process()
         addProcess.executableURL = URL(fileURLWithPath: "/usr/bin/security")
-        addProcess.arguments = ["add-generic-password", "-s", keychainService, "-w", jsonString]
+        addProcess.arguments = Self.keychainSaveArguments(
+            service: keychainService,
+            account: NSUserName(),
+            password: jsonString
+        )
         addProcess.standardOutput = Pipe()
         addProcess.standardError = Pipe()
 

--- a/Tests/InfrastructureTests/Claude/ClaudeCredentialLoaderTests.swift
+++ b/Tests/InfrastructureTests/Claude/ClaudeCredentialLoaderTests.swift
@@ -224,6 +224,31 @@ struct ClaudeCredentialLoaderTests {
         #expect(reloaded?.oauth.refreshToken == "new-refresh")
     }
 
+    @Test
+    func `keychain save arguments update existing item for account`() {
+        let password = """
+        {
+          "claudeAiOauth": {
+            "accessToken": "refreshed-token"
+          }
+        }
+        """
+
+        let arguments = ClaudeCredentialLoader.keychainSaveArguments(
+            service: "Claude Code-credentials",
+            account: "test-user",
+            password: password
+        )
+
+        #expect(arguments == [
+            "add-generic-password",
+            "-U",
+            "-s", "Claude Code-credentials",
+            "-a", "test-user",
+            "-w", password
+        ])
+    }
+
     // MARK: - Environment Variable Tests
 
     @Test

